### PR TITLE
yolo v2 changes

### PR DESCRIPTION
**Pull request by symphonyfi**
This PR includes the features of YOLO v2.

- [x] Test Case Passing

### CODE REVIEW

File: config/arguments.json [hunk 1]
```diff
@@ -1,9 +1,11 @@
 {
-  "symphony": {
+  "yolo": {
     "_comment::executionFee": "20 = 0.2%",
     "executionFee": 20,
-    "_comment::protocolFee": "2500 = 0.2 / 4 = 0.05%",
-    "protocolFee": 2500
+    "_comment::protocolFeePercent": "4000 = 0.2 * 0.4 = 0.08%",
```
Comment (symphonyfi) [inline config/arguments.json:5]:
```suggestion
    "_comment::protocolFeePercent": "4000 = 0.2 / (10000/4000) = 0.08%",
```

File: contracts/adapters/AaveYield.sol [hunk 2]
```diff
@@ -1,220 +1,140 @@
...
+    IAaveLendingPool public immutable lendingPool;
+    IAaveIncentivesController public immutable incentivesController;
+    address public immutable rewardToken;
+    uint16 internal constant referralCode = 43915;
+
+    // Addresses related to swap
+    address[] route;
+    IUniswapRouter public router;
+    IUniswapRouter public backupRouter;
+    uint256 public harvestGasUsage = 1000000;
```
Comment (symphonyfi) [inline contracts/adapters/AaveYield.sol:37]:
```suggestion
    uint256 public harvestMaxGas = 1000000; // 1000k wei
```

File: contracts/adapters/AaveYield.sol [hunk 3]
```diff
@@ -1,220 +1,140 @@
...
-     * @dev Used to approve max token and add token for reward
-     * @param asset the address of token
-     **/
-    function maxApprove(address asset) external override onlySymphony {
-        require(
-            aAsset == address(0),
-            "AaveYield: Asset can't be changed after initilization."
-        );
+        // reimburse function caller
+        uint256 reimbursementAmt = harvestGasUsage * tx.gasprice;
```
Comment (symphonyfi) [inline contracts/adapters/AaveYield.sol:124]:
```suggestion
        uint256 reimbursementAmt = harvestMaxGas * tx.gasprice;
```

File: contracts/adapters/AaveYield.sol [hunk 4]
```diff
@@ -223,205 +143,128 @@ contract AaveYield is IYieldAdapter, Initializable {
...
+                _route[_route.length - 1] == tokenAddress,
+            "AaveYield::updateRoute: Incorrect route"
         );
+        route = _route;
+    }
 
-        lendingPool = _lendingPool;
-        protocolDataProvider = _protocolDataProvider;
+    function updateHarvestGas(uint256 _gas) external onlyManager {
+        harvestGasUsage = _gas;
```
Comment (symphonyfi) [inline contracts/adapters/AaveYield.sol:231]:
```suggestion
        harvestMaxGas = _gas;
```

File: contracts/interfaces/IAaveLendingPool.sol [hunk 5]
```diff
@@ -1,5 +1,5 @@
 // SPDX-License-Identifier: agpl-3.0
-pragma solidity 0.7.4;
+pragma solidity 0.8.10;
 pragma experimental ABIEncoderV2;
```
Comment (symphonyfi) [inline contracts/interfaces/IAaveLendingPool.sol:3]: remove this

File: contracts/adapters/AaveYield.sol [hunk 6]
```diff
@@ -1,427 +1,267 @@
...
-    uint256 public previousAccRewardPerShare;
-
-    mapping(bytes32 => uint256) public orderRewardDebt;
-
-    modifier onlySymphony() {
+    address internal immutable tokenAddress;
+    address internal immutable aTokenAddress;
+    IAaveLendingPool public immutable lendingPool;
+    IAaveIncentivesController public immutable incentivesController;
+    address public immutable rewardToken;
```
Comment (justAvariable) [inline contracts/adapters/AaveYield.sol:28]:
```suggestion
    address public immutable rewardToken;  // can only be native token i.e WMATIC
```

**Review by justAvariable — APPROVED** [review]

**Review by justAvariable — APPROVED** [review]

**Pull request merged**